lib/tapioca/runtime/reflection.rb

```diff
@@ -174,14 +175,24 @@ def resolve_loc(locations)
         resolved_loc.absolute_path || ""
       end
 
-      sig { params(singleton_class: Module).returns(T.nilable(Module)) }
-      def attached_class_of(singleton_class)
-        # https://stackoverflow.com/a/36622320/98634
-        result = ObjectSpace.each_object(singleton_class).find do |klass|
-          singleton_class_of(T.cast(klass, Module)) == singleton_class
+      # On Ruby 3.2 or newer, Class defines an attached_object method that returns the
+      # attached class of a singleton class without iterating ObjectSpace. On older
+      # versions of Ruby, we fall back to iterating ObjectSpace.
+      if CLASS_ATTACHED_OBJECT
```

Comment:
Nice, I guess this is working. Now the linter is unhappy and I guess you have two choices:
1. Move the common sig out of the `if` block. I think it should work just fine with Sorbet, but not sure.
2. Suppress the linting rule for this part of the codebase.

Reply:
Yeah Sorbet seems fine with it! I didn't realize you could do that.

Reply:
Oh, sorry, one more thing: Seen as the condition is being checked only once, WDYT about eliminating the constant and moving the condition inside the `if`:
```suggestion
      if Class.method_defined?(:attached_object)
```